### PR TITLE
Jira - ASPL 542 (Download Scan Log File)

### PR DESCRIPTION
**Pull request by mvjainhcl**

### CODE REVIEW

**Comment by mvjainhcl** [comment]
@mattmurp  - Please could you review this pull request ?

File: src/main/java/com/hcl/appscan/sdk/results/CloudResultsProvider.java [hunk 1]
```diff
@@ -237,4 +237,49 @@ protected String getReportStatus(String reportId) throws IOException, JSONExcept
...
+			return;
+		}
+		String request_url = authProvider.getServer() + String.format(API_SCANS_SCANLOGS, executionId);
+		Map<String, String> request_headers = authProvider.getAuthorizationHeader(true);
+		request_headers.put(CONTENT_LENGTH, "0"); //$NON-NLS-1$
+
+		HttpClient client = new HttpClient(m_scanProvider.getAuthenticationProvider().getProxy(),m_scanProvider.getAuthenticationProvider().getacceptInvalidCerts());
+		HttpResponse response = client.get(request_url, request_headers, null);
+
+		if (response.getResponseCode() == HttpsURLConnection.HTTP_OK) {
```
[the remark this answers is not in the document]
Comment (mvjainhcl) [reply src/main/java/com/hcl/appscan/sdk/results/CloudResultsProvider.java:266]: Got it. Changes done. Will push new code.

File: src/main/java/com/hcl/appscan/sdk/results/CloudResultsProvider.java [hunk 2]
```diff
@@ -237,4 +237,49 @@ protected String getReportStatus(String reportId) throws IOException, JSONExcept
...
+			if (destination.isDirectory()) {
+				String fileName = "ScanLog" + "_" + SystemUtil.getTimeStamp() + "." + "zip"; //$NON-NLS-1$ //$NON-NLS-2$
+				destination = new File(destination, fileName);
+			}
+
+			destination.getParentFile().mkdirs();
+			response.getResponseBodyAsFile(destination);
+		} else {
+			JSONObject object = (JSONObject) response.getResponseBodyAsJSON();
+			if (object.has(MESSAGE)) {
```
[the remark this answers is not in the document]
Comment (mvjainhcl) [reply src/main/java/com/hcl/appscan/sdk/results/CloudResultsProvider.java:276]: Understood. Changes done and will push the new code.

File: src/main/java/com/hcl/appscan/sdk/results/CloudResultsProvider.java [hunk 3]
```diff
@@ -237,4 +237,49 @@ protected String getReportStatus(String reportId) throws IOException, JSONExcept
     	JSONObject obj = (JSONObject) response.getResponseBodyAsJSON();
     	return obj.getString(STATUS);
     }
+
+
+	public void getScanLogFile(File file , String executionId) {
```
[the remark this answers is not in the document]
Comment (mvjainhcl) [reply src/main/java/com/hcl/appscan/sdk/results/CloudResultsProvider.java:242]: Yes there are checks in place. Actually the user wont supply the filename. The CLI code by default ,  will supply the filename in this format - "ScanLog" + "_" + SystemUtil.getTimeStamp() + "." + "zip"

**Comment by mvjainhcl** [comment]
@mattmurp  - Thanks a lot for the PR review comments. I've made the changes to the code as per your feedback and pushed the new changes. Please review and approve. Thanks.

**Review by mattmurp — APPROVED** [review]

**Comment by mattmurp** [comment]
Hi @mvjainhcl I approved the changes, but there is a merge conflict that needs to be resolved. After that is resolved, please let me know, as I believe that will require another approval.

**Comment by mvjainhcl** [comment]
Hi @mattmurp - I've resolved the merge conflicts.

**Pull request merged**